syntax = "proto3";

package mealprep;

import "user.proto";
import "recipe.proto";
import "shopping.proto";

service MealPrepService {
  rpc GenerateWeeklyMeals (GenerateWeeklyMealsRequest) returns (stream WeeklyMealPlanProgress);
  rpc RegenerateRecipe (RegenerateRecipeRequest) returns (Recipe);
  rpc ModifyRecipe (ModifyRecipeRequest) returns (Recipe);
  rpc GetShoppingList (GetShoppingListRequest) returns (ShoppingList);
}

message GenerateWeeklyMealsRequest {
  UserProfile user_profile = 1;
  WeeklyMealRequest weekly_meal_request = 2;
}

message WeeklyMealRequest {
  // Add fields as needed for weekly meal planning
}

message WeeklyMealPlanProgress {
  // Add fields for streaming progress updates
}

message RegenerateRecipeRequest {
  string recipe_id = 1;
  string regeneration_reason = 2;
}

message ModifyRecipeRequest {
  string recipe_id = 1;
  string modification_instructions = 2;
}

message GetShoppingListRequest {
  string weekly_meal_plan_id = 1;
  repeated PantryItem pantry = 2;
}
